src/components/TimeControl/TimeControlPicker.tsx

```diff
@@ -18,54 +18,66 @@
 import * as React from "react";
 import * as data from "data";
 import { _ } from "translate";
-import { TimeControl } from "./TimeControl";
+
 import { time_options, makeTimeControlParameters } from "./util";
 import { computeAverageMoveTime } from "goban";
-//import {errorAlerter} from 'misc';
+import { TimeControl, TimeControlTypes } from "./TimeControl";
+
+type TimeControlSystem = TimeControlTypes.TimeControlSystem;
 
 interface TimeControlPickerProperties {
     value?: TimeControl;
     onChange?: (tc: TimeControl) => void;
+    force_system?: TimeControlSystem;
 }
 
-// TODO: add TimeControl as the state for this component.
-// However, it will require some extra TLC. There are many checks like this:
-//     if (this.state.system === "fischer") { doSomething(this.state.initial_time; }
-// that TypeScript would have a better time with if we were a little more
-// precise in what properties we're looking for:
-//     if ("initial_time" in this.state) { doSomething(this.state.initial_time; }
 export class TimeControlPicker extends React.PureComponent<
     TimeControlPickerProperties,
-    any /*TimeControl*/
+    TimeControl
 > {
     time_control: TimeControl;
 
     constructor(props) {
         super(props);
 
         const speed = data.get("time_control.speed", "correspondence") || "correspondence";
-        const system = data.get("time_control.system", "fischer") || "fischer";
+        const system =
+            this.props.force_system ||
+            ((data.get("time_control.system", "fischer") || "fischer") as TimeControlSystem);
 
         this.state = Object.assign(
             recallTimeControlSettings(speed, system),
             this.props.value || {},
         );
-        if (this.state.time_control) {
-            (this.state as any).system = this.state.time_control;
-        }
+
         this.state = Object.assign(this.state, makeTimeControlParameters(this.state));
         this.time_control = makeTimeControlParameters(this.state);
     }
+
     UNSAFE_componentWillReceiveProps(next_props: TimeControlPickerProperties) {
-        let update: any = {};
-        if (this.props.value !== next_props.value) {
-            update = Object.assign(update, makeTimeControlParameters(next_props.value));
-        }
-        if (Object.keys(update).length) {
-            this.time_control = makeTimeControlParameters(update);
-            this.setState(update);
+        let update: TimeControl;
+        // console.log("Timepicker props", next_props);
+        if (
+            this.props.value !== next_props.value ||
+            this.props?.force_system !== this.state.system
+        ) {
+            if (next_props.force_system) {
+                update = Object.assign(
+                    this.state,
+                    recallTimeControlSettings(this.state.speed, next_props.force_system),
+                );
+            } else if (next_props.value) {
+                update = { ...makeTimeControlParameters(next_props.value) };
```

Comment:
Also here... we *don't* want to update from `next_props.value` if `force_system` exists?

Reply:
Pfff... you are right, that is not the intent, force_system should be letting through next_props.  I think this only works by luck of timing.    Ay yah yay yah!

Reply:
Actually, the way it "works" is luck in the sense that `value` and `force` were never asserted together in the scenarios I tried.   However, there is nothing that rules this out, so a fix is needed.

Reply:
OK - fix coming.  I can see now why it seemed unlikely that `value` and `force` would ever be applied together, because value is used in direct challenges and tournies and `force` is used in rengo which doesn't support direct or tournies... today!

Reply:
Sounds like tomorrow's problem ;) haha

Reply:
Ahh got it, the way it's used currently, it's only one or the other.  Well thanks for the fix, that looks good and will probably help if someone tries to do both down the line!